roles/openshift_hosted/tasks/router/router.yml

```diff
@@ -10,7 +10,7 @@
 
 - name: set_fact replicas
   set_fact:
-    replicas: "{{ openshift.hosted.router.replicas | default(router_nodes.results.results[0]['items'] | length) }}"
+    replicas: "{{ openshift.hosted.router.replicas|default(None) | get_router_replicas(router_nodes) }}"
```

Comment:
I'm just thinking that filters are meant to translate data, and translating from a value that is either an `int` or `None` to an `int` seems a bit odd and could be confusing for someone not already familiar with the code.